mempool/cat/pool.go

```diff
@@ -657,29 +667,40 @@ func (txmp *TxPool) recheckTransactions() {
 	txmp.logger.Debug(
 		"executing re-CheckTx for all remaining transactions",
 		"num_txs", txmp.Size(),
-		"height", txmp.height,
+		"height", txmp.Height(),
 	)
 
+	// Collect transactions currently in the mempool requiring recheck.
+	wtxs := txmp.store.getAllTxs()
+
 	// Issue CheckTx calls for each remaining transaction, and when all the
 	// rechecks are complete signal watchers that transactions may be available.
-	txmp.store.iterateOrderedTxs(func(wtx *wrappedTx) bool {
-		// The response for this CheckTx is handled by the default recheckTxCallback.
-		rsp, err := txmp.proxyAppConn.CheckTxSync(abci.RequestCheckTx{
-			Tx:   wtx.tx.Tx,
-			Type: abci.CheckTxType_Recheck,
-		})
-		if err != nil {
-			txmp.logger.Error("failed to execute CheckTx during recheck",
-				"err", err, "key", fmt.Sprintf("%x", wtx.key()))
-		} else {
-			txmp.handleRecheckResult(wtx, rsp)
+	go func() {
```

Comment:
what is this part doing? async rechecktx?

Reply:
yeah, part of the revert, so this is what we used to do until recently. without this, we hit mutex issues and nodes halt when using CAT